### PR TITLE
Adjust reference input styles

### PR DESCRIPTION
**Pull request by annano**
# Contribution checklist

## Brief description
Screenshot from 2023-10-09 20-30-25

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: packages/text-editor/src/components/ReferenceInput.svelte [hunk 1]
```diff
@@ -156,195 +157,133 @@
...
-        textFormatCategories={[
-          TextFormatCategory.TextDecoration,
-          TextFormatCategory.Link,
-          TextFormatCategory.List,
-          TextFormatCategory.Quote,
-          TextFormatCategory.Code
-        ]}
-      />
+<div class="ref-container" class:focusable={kind === 'accented'}>
+  {#if showHeader && $$slots.header}
```
Comment (lexiv0re) [inline packages/text-editor/src/components/ReferenceInput.svelte:163]: Why do we need `showHeader` here? Isn't missing `header` in `$$slots` already provides the same condition?

File: packages/text-editor/src/components/ReferenceInput.svelte [hunk 2]
```diff
@@ -156,195 +157,133 @@
...
-        color: var(--theme-trans-color);
-        cursor: not-allowed;
-      }
-    }
-  }
   .ref-container {
     display: flex;
     flex-direction: column;
     min-height: 4.5rem;
+    border-width: 0.0625rem;
```
Comment (lexiv0re) [inline packages/text-editor/src/components/ReferenceInput.svelte]: Minor: use `border` short syntax for the first three styles?

File: packages/text-editor/src/components/StyledTextEditor.svelte [hunk 3]
```diff
@@ -15,23 +15,31 @@
...
+  } from '@hcengineering/ui'
   import { createEventDispatcher } from 'svelte'
   import textEditorPlugin from '../plugin'
   import { RefInputAction, RefInputActionItem, TextEditorHandler, TextFormatCategory } from '../types'
   import Attach from './icons/Attach.svelte'
   import { AnyExtension, mergeAttributes } from '@tiptap/core'
-  import StyleButton from './StyleButton.svelte'
   import TextEditor from './TextEditor.svelte'
   import { Node as ProseMirrorNode } from '@tiptap/pm/model'
+  import Button from '@hcengineering/ui/src/components/Button.svelte'
```
Comment (lexiv0re) [inline packages/text-editor/src/components/StyledTextEditor.svelte]: Move to import from `@hcengineering/ui`?

File: packages/text-editor/src/components/StyledTextEditor.svelte [hunk 4]
```diff
@@ -300,16 +315,16 @@
...
       }
+    }
 
-      &.focusable {
-        margin: -0.25rem -0.5rem;
-        padding: 0.25rem 0.5rem;
-        border: 1px solid transparent;
-        border-radius: 0.25rem;
+    &.focusable {
+      border-width: 0.0625rem;
```
Comment (lexiv0re) [inline packages/text-editor/src/components/StyledTextEditor.svelte]: Minor: use shorthand `border` style?

File: packages/text-editor/src/components/ReferenceInput.svelte
Comment (lexiv0re) [inline packages/text-editor/src/components/ReferenceInput.svelte:1]: `Description` is now shifted by the left padding in the New Issue dialog which doesn't look good:
NewIssueDialog

File: packages/text-editor/src/components/StyledTextEditor.svelte (hunk 4 above)
Comment (annano) [reply packages/text-editor/src/components/StyledTextEditor.svelte]: I want to update only border color here on `focus-within`, that is why I am using it like this

Comment (lexiv0re) [reply packages/text-editor/src/components/StyledTextEditor.svelte]: It doesn't mean you cannot use the shorthand property outside of `focus-within`

Comment (lexiv0re) [reply packages/text-editor/src/components/StyledTextEditor.svelte]: E.g. the following works:
```
.x {
  border: 1px solid gray;
  
  &.x_y {
    border-color: red;
  }
```

File: packages/text-editor/src/components/ReferenceInput.svelte [hunk 5]
```diff
@@ -156,195 +157,133 @@
...
+      on:content={(ev) => {
+        if (!isEmpty || haveAttachment) {
+          dispatch('message', ev.detail)
+          content = ''
+          textEditor.clear()
+        }
+      }}
+      on:blur={() => {
+        focused = false
+        dispatch('blur', focused)
```
Comment (aonnikov) [inline packages/text-editor/src/components/ReferenceInput.svelte]: I know that it is existing code, but `focused` here and in `focus` event is always the same value. Can we remove it? I searched through code and it looks like we mostly use `blur` and `focus` events without parameters.

File: packages/text-editor/src/components/ReferenceInput.svelte (hunk 1 above)
Comment (aonnikov) [reply packages/text-editor/src/components/ReferenceInput.svelte:163]: I think it is mostly for consistency with `showActions` and `showSend`. The `header` slot can be just a part of the header and even when it is not provided we may want to render some header content. But I'm ok with the proposed change as well.

File: packages/text-editor/src/components/ReferenceInput.svelte [hunk 6]
```diff
@@ -156,195 +157,133 @@
...
-          }
-        }
+  .header {
+    padding: 0.325rem 0.75rem;
+    border-bottom: 0.0625rem solid var(--theme-refinput-border);
+  }
 
-        &:disabled {
-          pointer-events: none;
+  .textInput {
```
Comment (aonnikov) [inline packages/text-editor/src/components/ReferenceInput.svelte]: Can we rename it to `text-input`?

File: packages/text-editor/src/components/ReferenceInput.svelte (hunk 1 above)
Comment (annano) [reply packages/text-editor/src/components/ReferenceInput.svelte:163]: It is mostly to manage bottom border and paddings for header component, I think it is better to manage it inside the component itself than in slot

File: packages/text-editor/src/components/ReferenceInput.svelte (hunk 6 above)
Comment (annano) [reply packages/text-editor/src/components/ReferenceInput.svelte]: Updated

File: packages/text-editor/src/components/ReferenceInput.svelte (hunk 5 above)
Comment (annano) [reply packages/text-editor/src/components/ReferenceInput.svelte]: Deleted

File: packages/text-editor/src/components/ReferenceInput.svelte
Comment (annano) [reply packages/text-editor/src/components/ReferenceInput.svelte:1]: Reverted

File: packages/text-editor/src/components/StyledTextEditor.svelte (hunk 4 above)
Comment (annano) [reply packages/text-editor/src/components/StyledTextEditor.svelte]: Updated

File: packages/text-editor/src/components/StyledTextEditor.svelte (hunk 3 above)
Comment (annano) [reply packages/text-editor/src/components/StyledTextEditor.svelte]: Updated

File: packages/text-editor/src/components/ReferenceInput.svelte (hunk 2 above)
Comment (annano) [reply packages/text-editor/src/components/ReferenceInput.svelte]: Updated

**Review by lexiv0re — APPROVED** [review]

**Review by aonnikov — APPROVED** [review]

**Pull request merged**